src/onegov/search/mixins.py

```diff
@@ -134,6 +134,122 @@ def es_tags(self):
         """ Returns a list of tags associated with this content. """
         return None
 
+    @property
+    def search_score(self):
+        """
+        the lower the score they higher the class type will be shown in search
+        results. Default is 10 (lowest)
+        """
+        return 10
+
+    def __lt__(self, other):
+        return self.search_score < other.search_score
+
+    @staticmethod
+    def psql_tsvector_string():
+        """
+        Provides the tsvector string for postgres defining which columns and
+        json fields to be used for full text search index.
+
+        Example:
+        s = create_tsvector_string('title', 'location')
+        s += " || ' ' || coalesce(((content ->> 'description')), '')"
+        s += " || ' ' || coalesce(((content ->> 'organizer')), '')"
+
+        :return: tsvector string
+        """
+        raise NotImplementedError
+
+    @staticmethod
+    def reindex(session, schema, model):
+        """
+        Re-indexes the table by dropping and adding the full text search
+        column.
+        """
+        Searchable.drop_fts_column(session, schema, model)
+        Searchable.add_fts_column(session, schema, model)
```

Comment:
Since we are doing that anyways, do we even need the upgrade step? It seems simpler to just ignore missing columns in `drop_fts_column`.

Reply:
Good point, we can skip the upgrade step. I will test it!

Reply:
You also might need to be more careful about polymorphism here, so you don't reindex the same table multiple times, since more than one Model can share the same table. So you might want to factor this a little differently.